test/browser/fixtures/locators/query.test.ts

```diff
@@ -13,4 +13,147 @@ test('can find elements inside the body', () => {
   document.body.innerHTML = '<div><span>hello</span></div>'
   const screen = page.elementLocator(document.body)
   expect(screen.getByText('hello').element()).toBe(document.querySelector('span'));
-})
\ No newline at end of file
+})
+
+describe('locator.and', () => {
+  test('can find a button matching multiple criteria', () => {
+    document.body.innerHTML = `
+    <button title="Testing framework">Vitest</button>
+    `
+    const locator = page.getByRole('button').and(page.getByTitle('Testing framework'))
+    expect(locator.element()).toBe(document.querySelector('button'))
+  })
+
+  test('throws an error if matches multiple elements', () => {
+    document.body.innerHTML = `
+    <button title="Testing framework">Vitest</button>
+    <button title="Testing framework">Vitest</button>
+    `
+    const locator = page.getByRole('button').and(page.getByTitle('Testing framework'))
+    expect(() => locator.element()).toThrowError(
+      `strict mode violation: getByRole('button').and(getByTitle('Testing framework')) resolved to 2 elements`
+    )
+  })
+})
+
+describe('locator.or', () => {
+  test('can find a button matching one of the criteria', () => {
+    document.body.innerHTML = `
+    <button title="Testing framework">Vitest</button>
+    `
+    const locator1 = page.getByRole('button').or(page.getByTitle('Non-existing title'))
+    expect(locator1.element()).toBe(document.querySelector('button'))
+
+    const locator2 = page.getByTitle('Testing framework').or(page.getByRole('alertdialog'))
+    expect(locator2.element()).toBe(document.querySelector('button'))
+  })
+
+  test('can find a button matching both criteria', () => {
+    document.body.innerHTML = `
+    <button title="Testing framework">Vitest</button>
+    `
+    const locator = page.getByRole('button').or(page.getByTitle('Testing framework'))
+    expect(locator.element()).toBe(document.querySelector('button'))
+  })
+
+  test('throws an error if matches multiple elements', () => {
+    document.body.innerHTML = `
+    <button title="Testing framework">Vitest</button>
+    <button title="Testing framework">Vitest</button>
+    `
+    const locator = page.getByRole('button').or(page.getByTitle('Testing framework'))
+    expect(() => locator.element()).toThrowError(
+      `strict mode violation: getByRole('button').or(getByTitle('Testing framework')) resolved to 2 elements`
+    )
+  })
+})
```

Comment:
Doesn't this test fail even without `or()`? I think we need a test where it fails explicitly due to `or` matching multiple elements, like:


```ts
    document.body.innerHTML = `
    <button>Vitest</button>
    <a href="https://vitest.dev" title="Testing framework">Vitest</a>
    `

    const locator = page.getByRole('button').or(page.getByTitle('Testing framework'))
```